### PR TITLE
NASA land satellite imagery plugin

### PR DESCRIPTION
**Pull request by nospampleasemam**
**What does your Instant Answer do?**

shows satellite imagery for a given lat/lon

**What problem does your Instant Answer solve (Why is it better than organic links)?**

organic links do not show satellite imagery for a given lat/lon:
https://duckduckgo.com/?q=landsat+40.7033127%2C-73.979681

**What is the data source for your Instant Answer? (Provide a link if possible)**

https://data.nasa.gov/developer/external/planetary/#imagery

**Why did you choose this data source?**

NASA is cool.

**Are there any other alternative (better) data sources?**

Not that I'm aware of.

**What are some example queries that trigger this Instant Answer?**

https://duckduckgo.com/?q=landsat+40.7033127%2C-73.979681

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**

People who like space. Geographers.

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**

No.

**Which existing Instant Answers will this one supersede/overlap with?**

None.

**Are you having any problems? Do you need our help with anything?**

No.

**What are the terms of use for the API? Will DuckDuckGo need specific authorization (e.g. an API key)? Are there any costs associated with API usage?**

Yes: https://data.nasa.gov/developer/external/planetary/#apply-for-an-api-key

1k hourly limit default / could be raised

`DEMO_KEY` api_key can be used for development

**Where did you hear about DuckDuckHack? (For first time contributors)**

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

http://imgur.com/h1cfHAl

## Checklist
Please place an 'X' where appropriate.

```
[X] Added metadata and attribution information
[X] Wrote test file and added to t/ directory
[] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
[] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
[] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 10

    - Windows 7
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 8
        [] IE 9
        [] IE 10

    - Windows XP
        [] IE 8

    - Mac OSX
        [] Google Chrome
        [] Firefox
        [] Opera
        [] Safari

    - iOS (iPhone)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - iOS (iPad)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - Android
        [] Firefox
        [] Native Browser
        [] Google Chrome
        [] Opera
```

### CODE REVIEW

**Comment by moollaza** [comment]
@nospampleasemam thanks a lot for contributing, we really appreciate it! We'll be sure to review this shortly and give you some feedback!

File: lib/DDG/Spice/Landsat.pm [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+name 'Landsat';
+attribution web => ['http://dylansserver.com','Dylan Lloyd'],
+            email => ['dylan@dylansserver.com','Dylan Lloyd'],
+            github => ['https://www.github.com/nospampleasemam', 'Dylan Lloyd'];
+
+spice to => 'https://api.data.gov/nasa/planetary/earth/imagery?api_key={{ENV{DDG_SPICE_NASA_APIKEY}}}&lat=$1&lon=$2&dim=0.5';
+
+spice wrap_jsonp_callback => 1;
+
+triggers query_lc => qr/^
```
Comment (moollaza) [inline lib/DDG/Spice/Landsat.pm]: Can we switch this to use trigger words and a regex guard? Would be a little more efficient!

**Comment by moollaza** [comment]
Let's ditch the custom template and utilize our built-in media template group:

```js
(function(env){
    "use strict";
    env.ddg_spice_landsat = function(api_result){

        if (!api_result.url) {
            return Spice.failed('landsat');
        }

        Spice.add({
            id: 'landsat',
            data: api_result,
            name: 'Landsat',
            meta: {
                sourceUrl: 'https://data.nasa.gov',
                sourceName: 'NASA',
                sourceIcon: true
            },
            normalize: function(data){
                return {
                    title: data.id,
                    subtitle: data.date,
                    image: data.url
                };
            },
            templates: {
                group: 'media',
                item: false,
                options: {
                    moreAt: true
                }
            }
        });
    };

}(this));
```

I'm struggling to think of a useful title/subtitle/description for this.

@abeyang any ideas on what we could put -- the API resposne is extremely minimal, the only other piece of info we get which we're not showing is the `planet:` param

**Comment by moollaza** [comment]
I was thinking of the lat/lon for the title and the subtitle as "Satellite Image".

We could maybe use the record template for the content area to show the id and time?

**Comment by javathunderman** [comment]
ping @nospampleasemam Are you interested in finishing this one up?

**Comment by moollaza** [comment]
@nospampleasemam sorry for the delay here -- do you know if the API is still around? I can't get the API calls to work..

**Comment by nospampleasemam** [comment]
NP, we talked about this on irc iirc. Unfortunately, it looks like the API has been discontinued. I sent an email to the project lead at NASA but did not receive a reply. Closing this PR now. :-(

**Comment by moollaza** [comment]
>NP, we talked about this on irc iirc. 

@nospampleasemam I thought so! Just wanted to see if there had been any changes. It sucks that they've deprecated the source...

**Pull request closed without merging**